chain/deals/provider.go

```diff
@@ -85,7 +85,7 @@ var (
 	ErrDataTransferFailed = errors.New("deal data transfer failed")
 )
 
-func NewProvider(ds dtypes.MetadataDS, sminer *storage.Miner, secb *sectorblocks.SectorBlocks, dag dtypes.StagingDAG, dataTransfer dtypes.ProviderDataTransfer, fullNode api.FullNode) (storagemarket.StorageProvider, error) {
+func NewProvider(ds dtypes.MetadataDS, sminer *storage.Miner, secb *sectorblocks.SectorBlocks, dag dtypes.StagingDAG, dataTransfer dtypes.ProviderDataTransfer, fullNode api.FullNode, spn storagemarket.StorageProviderNode) (storagemarket.StorageProvider, error) {
```

Comment:
also blocking: sminer and secb should not be needed? I hope. Either way we should remove them and if there are still dependencies lets identify them.